src/routes/routes.jsx

```diff
@@ -9,17 +9,40 @@ import {
   SignIn,
   SignUp,
 } from '../pages/all';
+import { useSelector } from 'react-redux';
+import { selectCurrentUserStatus } from '../features/auth/authSlice';
 
 export const RouterRoutes = () => {
+  const authStatus = useSelector(selectCurrentUserStatus);
+  console.log(authStatus);
+
   return (
     <Routes>
       <Route path='/' element={<Landing />} />
-      <Route path='/home' element={<Home />} />
-      <Route path='/explore' element={<Explore />} />
-      <Route path='/bookmarks' element={<Bookmarks />} />
-      <Route path='/profile' element={<Profile />} />
-      <Route path='/auth-signin' element={<SignIn />} />
-      <Route path='/auth-signup' element={<SignUp />} />
+      <Route
+        path='/home'
+        element={authStatus ? <Home /> : <Navigate to={'/auth/signin'} />}
+      />
```

Comment:
You can convert into a `RequireAuth` component.

Reply:
thanks for the suggestions , I have added those in my next pr 👍